txnprovider/txpool/txpoolcfg/txpoolcfg.go

```diff
@@ -212,68 +211,81 @@ func CalcIntrinsicGas(dataLen, dataNonZeroLen, authorizationsLen uint64, accessL
 
 		product, overflow := emath.SafeMul(nz, nonZeroGas)
 		if overflow {
-			return 0, GasUintOverflow
+			return 0, 0, GasUintOverflow
 		}
 		gas, overflow = emath.SafeAdd(gas, product)
 		if overflow {
-			return 0, GasUintOverflow
+			return 0, 0, GasUintOverflow
 		}
 
 		z := dataLen - nz
 
 		product, overflow = emath.SafeMul(z, fixedgas.TxDataZeroGas)
 		if overflow {
-			return 0, GasUintOverflow
+			return 0, 0, GasUintOverflow
 		}
 		gas, overflow = emath.SafeAdd(gas, product)
 		if overflow {
-			return 0, GasUintOverflow
+			return 0, 0, GasUintOverflow
 		}
 
 		if isContractCreation && isShanghai {
 			numWords := toWordSize(dataLen)
 			product, overflow = emath.SafeMul(numWords, fixedgas.InitCodeWordGas)
 			if overflow {
-				return 0, GasUintOverflow
+				return 0, 0, GasUintOverflow
 			}
 			gas, overflow = emath.SafeAdd(gas, product)
 			if overflow {
-				return 0, GasUintOverflow
+				return 0, 0, GasUintOverflow
+			}
+		}
+
+		// EIP-7623
+		if isPrague {
+			tokenLen := dataLen + 3*nz
+			floorGas7623, overflow = emath.SafeMul(tokenLen, fixedgas.TxTotalCostFloorPerToken)
+			if overflow {
+				return 0, 0, GasUintOverflow
+			}
+			floorGas7623, overflow = emath.SafeAdd(floorGas7623, fixedgas.TxGas)
+			if overflow {
+				return 0, 0, GasUintOverflow
 			}
 		}
 	}
```

Comment:
`floorGas7623` should be set to `fixedgas.TxGas` rather than 0 when `dataLen == 0` (probably immaterial, but we should still fix it for consistency).